Bugzilla/User.pm

```diff
@@ -366,6 +366,33 @@ sub update {
     $dbh->do("DELETE FROM profile_mfa WHERE user_id = ?", undef, $self->id);
   }
 
+  # IAM usernames are stored separately from normal profiles
+  # data so we update them here instead
+  my $new_iam_username = delete $self->{new_iam_username};
+  my $old_iam_username = $self->iam_username;
+
+  $dbh->bz_start_transaction();
+  if (!defined $old_iam_username && $new_iam_username) {
```

Comment:
We'd need a `VALIDATOR` for `iam_username` to ensure they match the format that IAM requires.

This would also alleviate my concerns about Perl's truthiness tests of strings (setting `iam_username` to `"0"` would break these tests).